src/transport_ofi.c

```diff
@@ -463,18 +420,27 @@ int shmem_transport_ofi_stx_search_shared(long threshold)
             break;
 
         case RANDOM:
-            shmem_transport_ofi_stx_rand_restart();
-            while ((i = shmem_transport_ofi_stx_rand_next(threshold)) >= 0) {
+            for (i = count = 0; i < shmem_transport_ofi_stx_max; i++) {
                 if (shmem_transport_ofi_stx_pool[i].ref_cnt > 0 &&
-                    (shmem_transport_ofi_stx_threshold_cmp(i, threshold) || threshold == -1) &&
-                    !shmem_transport_ofi_stx_pool[i].is_private) {
-                    stx_idx = i;
-                    rand_pool_prev_choice = stx_idx;
-                    rr_start_idx = (i + 1) % shmem_transport_ofi_stx_max;
-                    break;
+                    (shmem_transport_ofi_stx_pool[i].ref_cnt <= threshold || threshold == -1) &&
+                    !shmem_transport_ofi_stx_pool[i].is_private)
+                {
+                    ++count;
```

Comment:
Do we want a `break;` here?

Reply:
Oh, yeah, that's a nice optimization to break early.  IIUC, this shouldn't change the behavior of the allocator (apart from making it faster, which is always a good thing), right?

Reply:
Right!